src/graph/graph.c

```diff
@@ -1142,14 +1160,74 @@ void Graph_DeleteEdges
 			// there are no additional edges connecting source to destination
 			// remove edge from THE adjacency matrix
 			if(!connected) {
-				M = Graph_GetAdjacencyMatrix(g, false);
-				info = RG_Matrix_removeElement_BOOL(M, src_id, dest_id);
+				Delta_Matrix adj = Graph_GetAdjacencyMatrix(g, false);
+				info = Delta_Matrix_removeElement(adj, src_id, dest_id);
 				ASSERT(info == GrB_SUCCESS);
 			}
+		} else {
+			ME  = Graph_MultiEdgeRelationMatrix(g, r);
+			me_id = CLEAR_MSB(me_id);
+			info = Delta_Matrix_removeElement(ME, me_id, edge_id);
```

Comment:
What happens if `NVALS(ME[me_id:]) == 1` ? meaning we've transitioned from MULTI-EDGE to a single edge?
we need to remove the MSB in addition to add the `me_id` row to the matrix `freelist`, this is why I suggested encapsulating this operations for the multi-edge matrix.